### PR TITLE
[PO-79] Use block number to get the contract state

### PR DESCRIPTION
**Pull request by mirkoAlic**

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/governance/CertificateAuthorityManager.sol [hunk 1]
```diff
@@ -6,8 +6,8 @@ contract CertificateAuthorityManager {
 
 	address a1 = 0x00edde8656c35fcb7126c61fc6e2673734425a72bf;
 
-	function isCertificateAuthorityFor(address sender, uint slotNumber) constant returns(bool) {
-		CAQueriedFor(sender, slotNumber);
+	function isCertificateAuthorityFor(address sender) constant returns(bool) {
+		CAQueriedFor(sender, 0);
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/governance/CertificateAuthorityManager.sol:10]: Minor detail: Maybe we can remove the parameter from the event that will always be 0?

File: src/main/scala/io/iohk/ethereum/governance/CertificateAuthorityManager.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ import io.iohk.ethereum.vm.utils.{Contract, Utils}
 
 
 trait CertificateAuthorityManager {
-  def isCertificateAuthorityFor(address: Address, slotNumber: BigInt): Boolean
+  def isCertificateAuthorityFor(address: Address, lastBlockHeader: BlockHeader): Boolean
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/governance/CertificateAuthorityManager.scala]: Minor detail: Maybe we should add a comment regarding the fact that the state at which `lastBlockHeader` leaves the contract is used for executing the query?

**Review by ntallar — COMMENTED** [review]
Apart from two minor comments, LGTM!

File: src/main/scala/io/iohk/ethereum/governance/CertificateAuthorityManager.sol (hunk 1 above)
Comment (mirkoAlic) [reply src/main/scala/io/iohk/ethereum/governance/CertificateAuthorityManager.sol:10]: this will be fix in the next PR! =)

**Review by ntallar — APPROVED** [review]
LGTM!

**Pull request merged**